modules/transactionpool/accept.go

```diff
@@ -335,10 +335,24 @@ func (tp *TransactionPool) AcceptTransactionSet(ts []types.Transaction) error {
 		tp.mu.Lock()
 		defer tp.mu.Unlock()
 		err := tp.acceptTransactionSet(ts, txnFn)
+		// In case of certain errors we still want to broadcast the set
+		broadcast := err == nil || err == modules.ErrDuplicateTransactionSet
+		if broadcast {
+			// This set was broadcasted before if err != nil. We need to update
+			// it's seen txn height when we rebroadcast it. If we don't do
+			// that, the transaction will be pruned from the tpool and they
+			// might no longer show up in the wallet while still beingt
+			// tracked.
+			if err != nil {
+				for _, txn := range ts {
+					tp.transactionHeights[txn.ID()] = tp.blockHeight
+				}
+			}
+			go tp.gateway.Broadcast("RelayTransactionSet", ts, tp.gateway.Peers())
```

Comment:
calling `tp.gateway.Peers()` while the tpool is under lock is a deadlock risk, need to restructure to avoid this (looks like the existing code had this deadlock risk in it as well)

Reply:
The general convention is that a module can't call exported methods of other modules while under lock